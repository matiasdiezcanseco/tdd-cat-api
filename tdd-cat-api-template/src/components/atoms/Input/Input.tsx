interface IInput {
  onChange: (e: string) => void
  title?: string
}

const Input: React.FC<IInput> = ({}) => {
  return (
    <>
      <input className="border-[1px] rounded-md border-gray-600 p-2 bg-transparent" type="text" />
    </>
  )
}

export default Input
