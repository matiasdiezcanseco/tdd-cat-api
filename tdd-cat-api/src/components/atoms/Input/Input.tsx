interface IInput {
  onChange: (e: string) => void
  title?: string
}

const Input: React.FC<IInput> = ({ onChange, title = '' }) => {
  return (
    <>
      <label>{title}</label>
      <input
        className="border-[1px] rounded-md border-gray-600 p-2 bg-transparent"
        type="text"
        onChange={(e: any) => onChange(e.target.value)}
      />
    </>
  )
}

export default Input
