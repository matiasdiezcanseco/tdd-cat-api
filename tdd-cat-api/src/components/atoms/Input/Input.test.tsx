import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Input from './Input'
describe('test Input', () => {
  let onChange = jest.fn()
  beforeAll(() => {
    onChange = jest.fn()
    jest.clearAllMocks()
  })

  test('should show text value', async () => {
    render(<Input onChange={onChange} />)
    const textbox = screen.getByRole('textbox')
    await userEvent.type(textbox, 'test')
    expect(textbox).toHaveValue('test')
  })

  test('should call onChange with correct value', async () => {
    render(<Input onChange={onChange} />)
    const textbox = screen.getByRole('textbox')
    expect(onChange).toHaveBeenCalledTimes(0)
    await userEvent.type(textbox, 'test')
    expect(onChange).toHaveBeenCalledTimes(4)
    expect(onChange).toHaveBeenCalledWith('test')
  })

  test('should show title', () => {
    render(<Input onChange={onChange} title="title" />)
    expect(screen.getByText('title')).toBeVisible()
  })

  test('should match snapshot', () => {
    const { container } = render(<Input onChange={onChange} />)
    expect(container).toMatchSnapshot()
  })
})
