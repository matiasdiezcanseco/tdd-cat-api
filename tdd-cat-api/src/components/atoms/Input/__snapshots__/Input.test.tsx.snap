// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Input should match snapshot 1`] = `
<div>
  <label />
  <input
    class="border-[1px] rounded-md border-gray-600 p-2 bg-transparent"
    type="text"
  />
</div>
`;
